src/governance/Migrate.t.sol

```diff
@@ -0,0 +1,317 @@
+        uint[2] memory durations;
+        durations[0] = 30 days;
+        durations[1] = 5 * 30 days;
+
+        jarConnector = new JarConnector(address(manager), address(ethJoin), "ETH", durations);
+        score.transferOwnership(address(jarConnector));
+        jarConnector.spin();
+
+        migrate = new Migrate(jarConnector, manager);
+        timelock = new Timelock(address(migrate), 2 days);
+        migrate.setTimelock(timelock);
+
+        (users, cdps) = createUsers(10);
+        
+    }
+
+    function testValidateSetup() public {
+        assertEq(address(migrate.timelock()), address(timelock));
+        assertEq(timelock.admin(), address(migrate));
+        assertEq(timelock.delay(), 2 days);
+    }
+
+    function testFailSetTimelock() public {
+        Timelock fakeTimelock = new Timelock(address(migrate), 2 days);
+        migrate.setTimelock(fakeTimelock);
+    }
+
+    function testFailNewProposalBeforeSixMonths() public {
+        FakeUser user = new FakeUser();
+        migrate.propose(address(user));
+    }
+
+    function testNewProposal() public {
+        forwardTimeSixMonths();
+
+        FakeUser user = new FakeUser();
+        migrate.propose(address(user));
+
+        (uint forVotes, , address owner) = migrate.proposals(0);
+        assertEq(forVotes, 0);
+        assertEq(owner, address(user));
+    }
+
+    function testAllowMultipleProposal() public {
+        forwardTimeSixMonths();
+
+        // first proposal
+        FakeUser user1 = new FakeUser();
+        uint proposalId = migrate.propose(address(user1));
+
+        (uint forVotes, , address owner) = migrate.proposals(proposalId);
+        assertEq(forVotes, 0);
+        assertEq(owner, address(user1));
+
+        // second proposal
+        FakeUser user2 = new FakeUser();
+        proposalId = migrate.propose(address(user2));
+
+        (forVotes, , owner) = migrate.proposals(proposalId);
+        assertEq(forVotes, 0);
+        assertEq(owner, address(user2));
+    }
+
+    function testVote() public {
+        forwardTimeSixMonths();
+
+        FakeUser user = new FakeUser();
+        uint proposalId = migrate.propose(address(user));
+
+        (uint forVotes, ,) = migrate.proposals(proposalId);
+        assertEq(forVotes, 0);
+
+        for(uint i = 0; i < 10; i++) {
+            users[i].doVote(migrate, proposalId, cdps[i]);
+        }
+
+        (forVotes, ,) = migrate.proposals(proposalId);
+        assertTrue(forVotes > 0);
+    }
+
+    function testCancelVote() public {
+        forwardTimeSixMonths();
+
+        FakeUser user = new FakeUser();
+        uint proposalId = migrate.propose(address(user));
+
+        (uint forVotes, ,) = migrate.proposals(proposalId);
+        assertEq(forVotes, 0);
+
+        for(uint i = 0; i < 10; i++) {
+            users[i].doVote(migrate, proposalId, cdps[i]);
+        }
+
+        (forVotes, ,) = migrate.proposals(proposalId);
+        assertTrue(forVotes > 0);
+
+        for(uint i = 0; i < 10; i++) {
```

Comment:
would be better if not everyone would cancel, and then check the reminder is fine.